ngraph/src/ngraph/op/interpolate.cpp

```diff
@@ -111,143 +116,518 @@ namespace ngraph
+        }
+        return out_shape;
+    }
+
+    template <typename T>
+    std::vector<T> correct_pad(const std::vector<T>& p, std::size_t rank)
+    {
+        std::size_t pad_len = p.size();
+        if (pad_len == rank)
+        {
+            return p;
+        }
+
+        std::vector<T> result;
+
+        if (pad_len > rank)
+        {
+            result.insert(result.end(), p.begin(), p.begin() + rank);
+        }
+        else
+        {
+            result = p;
+            result.insert(result.end(), rank - pad_len, 0);
+        }
+
+        return result;
+    }
+
+    template <element::Type_t ET>
+    inline bool evaluate(const HostTensorVector& args,
+                         const HostTensorPtr& out,
+                         const op::v4::Interpolate::InterpolateAttrs& attrs)
+    {
+        using T = typename element_type_traits<ET>::value_type;
+        using ShapeCalcMode = ngraph::op::v4::Interpolate::ShapeCalcMode;
+
+        Shape input_shape{args[data_port]->get_shape()};
+        std::size_t input_rank = input_shape.size();
+
+        auto axes = get_axes_vector(args);
+
+        std::size_t num_of_axes = axes.size();
+
+        auto scales = get_scales_vector(args, input_shape, attrs, axes);
+
+        auto pads_begin = correct_pad(attrs.pads_begin, input_rank);
+        auto pads_end = correct_pad(attrs.pads_end, input_rank);
+
+        auto padded_input_shape = get_padded_input_shape(input_shape, pads_begin, pads_end);
+
+        std::vector<std::size_t> out_shape_vector;
+
+        if (attrs.shape_calculation_mode == ShapeCalcMode::scales)
+        {
+            out_shape_vector = shape_infer_with_scales(padded_input_shape, axes, scales);
+        }
+        else
+        {
+            auto target_shape = get_target_shape_vector(args, num_of_axes);
+            out_shape_vector =
+                shape_infer_with_target_shape(padded_input_shape, axes, target_shape);
+        }
+
+        Shape out_shape{out_shape_vector};
+
+        out->set_shape(out_shape);
+
+        std::vector<T> padded_input_data(shape_size(Shape{padded_input_shape}));
+
+        runtime::NDimArrayView<T> padded_array_view{padded_input_data.data()};
+        runtime::NDimArrayView<T> input_array_view{args[0]->get_data_ptr<ET>()};
+
+        std::vector<int64_t> limits_for_indices(input_rank);
+        for (std::size_t i = 0; i < input_rank; ++i)
+        {
+            limits_for_indices[i] = padded_input_shape[i] - 1;
+        }
+        auto index_vector = limits_for_indices;
+
+        runtime::NDimIndex limits{index_vector, limits_for_indices};
+        runtime::NDimRange index_range{limits};
+        for (const auto& index : index_range)
+        {
+            auto padded_index = index;
+            for (std::size_t i = 0; i < input_rank; ++i)
+            {
+                padded_index[i] += pads_begin[i];
+            }
+            padded_array_view[padded_index] = input_array_view[index];
```

Comment:
I think you are copying every data from no-padded tensor to padded tensor here. What value should be filled in padded position, zeros as default?

Reply:
> I think you are copying every data from no-padded tensor to padded tensor here. What value should be filled in padded position, zeros as default?

Yes, padded positions will be filled by zeros, and this filling is performed in line 501:
```c++
std::vector<T> padded_input_data(shape_size(Shape{padded_input_shape}), T{});
```

Reply:
> I know this is a draft, but not very sure and clear:
> Here "index" in this line seemed to pointer to padded tensor(from code above, padded_input_shape --> limits_for_indice --> limits -->index_range --> index). So input_array_view[index] may be out of memory for some iterator, because input_array_view is not padded memory.

Fixed.